yt/data_objects/data_containers.py

```diff
@@ -808,11 +815,64 @@ def create_firefly_object(
             )
             datadir = JSONdir
 
+        if match_any_particle_types is None:
+            # not specified, switching to (temporary) default
+            issue_deprecation_warning(
+                "match_any_particle_types wasn't specified. Its current default "
+                "value (True) will be changed to False in the future. "
+                "Pass an actual value (True or False) to silence this warning. ",
+                since="4.2",
+            )
+            # Note that all of the tests in test_firefly.py have this explicitly
+            # specified (and labeled) to avoid this warning. They will need to be
+            # changed when this behavior changes in the future
+            match_any_particle_types = True
+
```

Comment:
```suggestion
```
If we're not going to change the default in the future, we should remove the deprecation warning, right?